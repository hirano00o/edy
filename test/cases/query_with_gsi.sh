#!/bin/bash

SCRIPT_ROOT_DIR=$1
TEST_NAME=$(basename "$0" | sed "s/\..*//")

# aws dynamodb query --table-name User --index EmailIndex --key-condition-expression Email=:email \
#   --expression-attribute-values "{\":email\":{\"S\":\"charlie@example.com\"}}" --endpoint-url http://localhost:8000
CMD="edy q -t User -p charlie@example.com --idx EmailIndex --local 8000"

. "${SCRIPT_ROOT_DIR}"/helper.sh

run_such_query_helper