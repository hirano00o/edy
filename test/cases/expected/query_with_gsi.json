[
  {
    "Address": {
      "City": "Los Angeles",
      "State": "California"
    },
    "Age": 24,
    "Birthday": {
      "Day": 21,
      "Month": 3,
      "Year": 1996
    },
    "Email": "charlie@example.com",
    "ID": 4,
    "Name": "Charlie"
  }
]
